### PR TITLE
Use pathlib in test_deepreload

### PR DESCRIPTION
**Pull request by PhanatosZou**
See: #12515

### CODE REVIEW

File: IPython/lib/tests/test_deepreload.py [hunk 1]
```diff
@@ -17,11 +17,10 @@ def test_deepreload():
...
         with prepended_to_syspath(tmpdir):
-            tmpdirpath = Path(tmpdir)
-            with open(tmpdirpath / "A.py", "w") as f:
-                f.write("class Object(object):\n    pass\n")
-            with open(tmpdirpath / "B.py", "w") as f:
-                f.write("import A\n")
+            tmpdirpathA = Path(tmpdir + "A.py")
+            tmpdirpathB = Path(tmpdir + "B.py")
+            tmpdirpathA.write_text("class Object(object):\n    pass\n")
+            tmpdirpathB.write_text("import A\n")
```
Comment (Carreau) [inline IPython/lib/tests/test_deepreload.py:23]: That is incorrect and test are failing. Please check your work, I belive you want / instead fo +.

Comment (PhanatosZou) [reply IPython/lib/tests/test_deepreload.py:23]: Exactly! Thanks for the reminder. Will do ASAP.

**Comment by PhanatosZou** [comment]
Hi, sorry for the late. I'd like to follow up with this PR and wondering if anyone can give me some tips. Any response will be appreciate.

**Comment by Carreau** [comment]
Thanks for what you did, I was not able to followup with all the IPython PRs, and now this conflict. To try to reduce the number of open PRs I'm going to close this. 

Apologies,

**Pull request closed without merging**